lib/daru/io/importers/excelx.rb

```diff
@@ -0,0 +1,132 @@
+        # @param skiprows [Integer] Skips the first +:skiprows+ number of rows from the
+        #   sheet being parsed.
+        # @param skipcols [Integer] Skips the first +:skipcols+ number of columns from the
+        #   sheet being parsed.
+        # @param order [Boolean] Defaults to true. When set to true, first row of the
+        #   given sheet is used as the order of the Daru::DataFrame and data of
+        #   the Dataframe consists of the remaining rows.
+        # @param index [Boolean] Defaults to false. When set to true, first column of the
+        #   given sheet is used as the index of the Daru::DataFrame and data of
+        #   the Dataframe consists of the remaining columns.
+        #
+        #   When set to false, a default order (0 to n-1) is chosen for the DataFrame,
+        #   and the data of the DataFrame consists of all rows in the sheet.
+        #
+        # @return A +Daru::DataFrame+ imported from the given XLSX file and sheet
+        #
+        # @example Importing from a local file
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(15x7)>
+        #              Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        0          H          1        nil        nil New stock  2014-08-01        nil
+        #        1        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        2        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        3        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a local file without headers
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet, headers: false).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(16x7)>
+        #                   0           1          2          3          4          5        6
+        #        0      Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        1          H          1        nil        nil New stock  2014-08-01        nil
+        #        2        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        3        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        4        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a remote URL
```

Comment:
As the linkages are done dynamically, YARD won't be able to recognize the linkages. The simplest workaround I can think for this, is to include their respective hyperlinks to the `Daru::DataFrame#register_io_module` YARD comments. Like,

```ruby
module Daru
  class DataFrame  
    class << self
      # Links +Daru::DataFrame+ Import / Export methods to corresponding 
      # +Daru::IO+ Importer / Exporter classes.
      #
      # - {Daru::IO::Importers::CSV#initialize #from_csv method}
      # - {Daru::IO::Importers::Redis#initialize #from_redis method}
      # ...
      def register_io_module(function, instance)
          ...
      end
    end
  end
end
```

I think it's reasonable to expect users to open the DataFrame page to have a look at the Importers & Exporters, and find a list of them as shown below.
![image](https://user-images.githubusercontent.com/17109060/28212978-967f86be-68c1-11e7-8d75-6cac309c7e56.png)
Does this suffice?